File: other/merge_ranges.py
def merge_ranges(ranges):
    sorted_ranges = sorted(ranges)
    merged_ranges = []

    previous_range_start, previous_range_end = sorted_ranges[0]

    for current_range_start, current_range_end in sorted_ranges[1:]:
        if current_range_start <= previous_range_end:
            previous_range_end = max(current_range_end, previous_range_end)
        else:
            merged_ranges.append((previous_range_start, previous_range_end))
            previous_range_start, previous_range_end = \
                current_range_start, current_range_end

    merged_ranges.append((previous_range_start, previous_range_end))

    return merged_ranges

ranges = [(1, 2), (1, 5), (7, 8)]
print merge_ranges(ranges)

File: other/merge_ranges.js
function mergeRanges(ranges) {
    var sortedRanges = ranges.sort(function(a, b){
        if(a[0] === b[0]){
          return a[1] - b[1];
        }
        else  {
          return a[0] - b[0];
        }
      });
    var mergedRanged = [];

    return sortedRanges;
}

ranges = [[1, 5], [7, 8], [1, 2]];
console.log(mergeRanges(ranges));